### PR TITLE
[TACHYON-1400] Skip cache promote if the block is already in memory

### PR DESCRIPTION
**Pull request by miasantreble**
https://tachyon.atlassian.net/browse/TACHYON-1400

I also want add test coverage for this but I'm not sure how we could find out if the `promoteBlock` RPC is indeed made (or not)

### CODE REVIEW

File: clients/unshaded/src/main/java/tachyon/client/block/TachyonBlockStore.java [hunk 1]
```diff
@@ -226,6 +226,12 @@ public void promote(long blockId) throws IOException {
       // Nothing to promote
       return;
     }
+    for (BlockLocation loc : info.getLocations()) {
+      if (loc.getTierAlias().equals("mem")) {
```
Comment (calvinjia) [inline clients/unshaded/src/main/java/tachyon/client/block/TachyonBlockStore.java]: Is there a way to make this less hardcoded? (ie. what if the user only has ssd/hdd tiers) We can assume all workers have the same tier storage configurations.

**Comment by yupeng9** [comment]
@miasantreble, a unit test will be good to have. You can mContext.acquireWorkerClient, and verify it's not called.

**Comment by gpang** [comment]
I agree with @yupeng9 . I think a unit test demonstrating this improvement would be great!

**Comment by haoyuan** [comment]
Thanks for the patch, @miasantreble . Are you still working on this?

**Comment by miasantreble** [comment]
@haoyuan , not right now, closing this PR.
Will open another one once I get some time, thanks!

**Pull request closed without merging**